src/software/new_geom/segment.cpp

```diff
@@ -0,0 +1,51 @@
+#include "software/new_geom/segment.h"
+
+void Segment::setStart(Point o)
+{
+    start = o;
+}
+
+const Point& Segment::getStart() const
+{
+    return start;
+}
+
+void Segment::setEnd(Point o)
+{
+    end = o;
+}
+
+const Point& Segment::getEnd() const
+{
+    return end;
+}
+
+double Segment::length() const
+{
+    return (end - start).length();
+};
+
+double Segment::lengthSquared() const
+{
+    return (end - start).lengthSquared();
+}
+
+Segment Segment::reverse() const
+{
+    return Segment(end, start);
+};
+
+Vector Segment::toVector() const
+{
+    return end - start;
+};
+
+double Segment::slope() const
+{
+    return (end.y() - start.y()) / (end.x() - start.x());
+};
+
+bool Segment::operator==(const Segment& other) const
+{
+    return start == other.start && end == other.end;
+}
```

Comment:
Could we add unit tests for this class?

Reply:
Separate ticket exists (that I'm also assigned to) for creating segment unit tests. If there's agreement that I can combine them into this PR, I'll add tests. Until then, I'm going to try and finish this ticket first (primarily getting CI to pass and addressing outstanding review comments).

Reply:
Added unit tests for segment class.